building_sim_plugins/building_ignition_plugins/src/slotcar.cpp

```diff
@@ -109,40 +111,76 @@ void SlotcarPlugin::Configure(const Entity& entity,
   //executor->spin();
   dataPtr->init_ros_node(_ros_node);
 
-  joints[0] = model.JointByName(ecm, "joint_tire_left");
-  if (!joints[0])
-    RCLCPP_ERROR(dataPtr->logger(),
-      "Could not find tire for [joint_tire_left]");
-
-  joints[1] = model.JointByName(ecm, "joint_tire_right");
-  if (!joints[1])
-    RCLCPP_ERROR(dataPtr->logger(),
-      "Could not find tire for [joint_tire_right]");
-
-  // Initialise JointVelocityCmd / JointVelocity components for velocity control
-  for (const auto& joint : joints)
-  {
-    if (!ecm.EntityHasComponentType(joint,
-      components::JointVelocityCmd().TypeId()))
-      ecm.CreateComponent(joint, components::JointVelocityCmd({0}));
-    if (!ecm.EntityHasComponentType(joint,
-      components::JointVelocity().TypeId()))
-      ecm.CreateComponent(joint, components::JointVelocity({0}));
-  }
   // Initialize Pose3d component
   if (!ecm.EntityHasComponentType(entity, components::Pose().TypeId()))
     ecm.CreateComponent(entity, components::Pose());
   // Initialize Bounding Box component
   if (!ecm.EntityHasComponentType(entity,
     components::AxisAlignedBox().TypeId()))
-  {
     ecm.CreateComponent(entity, components::AxisAlignedBox());
+  // Initialize Linear/AngularVelocityCmd components to drive slotcar
+  if (!ecm.EntityHasComponentType(_entity,
+    components::LinearVelocityCmd().TypeId()))
+    ecm.CreateComponent(_entity, components::LinearVelocityCmd());
+  if (!ecm.EntityHasComponentType(_entity,
+    components::AngularVelocityCmd().TypeId()))
+    ecm.CreateComponent(_entity, components::AngularVelocityCmd());
+
+  // Keep track of when a payload is dispensed onto/ingested from slotcar
+  // Needed for TPE Plugin to know when to manually move payload via this plugin
+  if (!_ign_node.Subscribe("/item_dispensed", &SlotcarPlugin::item_dispensed_cb,
+    this))
+  {
+    std::cerr << "Error subscribing to topic [/item_dispensed]" << std::endl;
```

Comment:
It would still be able to subscribe to a topic that hasn't been advertised yet, so the order in which the publishers/subscribers advertise/subscribe to the topic shouldn't matter.